pkg/utils/util.go

```diff
@@ -55,6 +56,34 @@ func HomeDir() string {
 	return dir
 }
 
+// ParsePath returns an absolute path, parsing ~ . .. etc
+func ParsePath(path string) (string, error) {
+	if path == "~" {
+		path = HomeDir()
+	} else if strings.HasPrefix(path, "~/") {
+		path = filepath.Join(HomeDir(), path[2:])
+	} else if username, err := user.Current(); err == nil && strings.HasPrefix(fmt.Sprintf("%s/", path), fmt.Sprintf("~%s/", username.Username)) {
+		prefix := fmt.Sprintf("~%s/", username.Username)
+		if strings.HasSuffix(path, "/") {
+			// support ~user/
+			path = filepath.Join(HomeDir(), path[len(prefix):])
+		} else {
+			// support ~user
+			path = filepath.Join(HomeDir(), path[len(prefix)-1:])
+		}
+	} else if strings.HasPrefix(path, "~") {
+		// we don't support ~otheruser/
+		return "", fmt.Errorf("unable to parse path, please specify an absolute path (e.g. /home/%s)", path[1:])
+	}
+
+	absolutePath, err := filepath.Abs(filepath.Clean(path))
+	if err != nil {
+		return "", err
+	}
+
+	return absolutePath, nil
+}
+
```

Comment:
Maybe a cleaner way to write this:

``` js
function parsePath(input) {
  if(input.hasPrefix("~")) {
    const first_path = input.split("/")[0]

    if(first_path !== "~" && first_path !== `~${USERNAME}`) {
      throw new Error("We don't support relative paths to point to other users!")
    }

    input.replace(first_path, ABSOLUTE_HOME_DIR_PATH)
  }

  return input
}
```

Reply:
Re-implemented using your implementation